docs/software-architecture-and-design.md

```diff
@@ -613,80 +601,26 @@ Now we have a nice loop from the `Simulator -> Sensor Fusion -> Validation Funct
-
-### Build A Hierarchy
-Qt is designed to handle hierarchy. It has [an extensive and robust system for maintaining Object Trees and parent/child relationships between components](https://doc.qt.io/qt-5/objecttrees.html). Much like regular code, GUIs should be created in a logical hierarchy to make organization and re-use easier.
-
-Make sure to use layouts, and group widgets in a logical way. For example, several widgets that work together to collect a user's mailing address should be grouped. This group may then be part of a larger group of widgets in a popup dialog asking for billing information.
-
-Here is a good example of laying out widgets in a hierarchy:
-
-![Good Qt Widget Layout](images/qt_layout_good_example.png)
-
-Do **not** lay out all your widgets in a single layer like this:
-
-![Bad Qt Widget Layout](images/qt_layout_bad_example.png)
-
-The main point to remember is to use [layouts](https://doc.qt.io/qt-5/layout.html) to group and manage widgets, and to create these groups in a logical way that can be built into a hierarchy. This will make it significantly easier to replace parts of the GUI later, or move components around.
-
-### Create Reusable Widgets
-Much like how we create functions in order to reuse code, [widgets](https://doc.qt.io/qt-5/qtwidgets-index.html) should be created so that they are reusable.
-
-For example, if you create a few widgets that work together to gather user input with a slider and display the current value next to it, you should combine all of this into its own `SliderWithValue` widget. This will make it very easy to make several copies of this widget, or use it somewhere else in a completely different application. Similarly, if you need specialized functionality from any widget (for example, our `ZoomableQGraphicsView`), this should also be implemented as a custom reusable widget.
-
-![Reusable Widget Example](images/qt_reusable_widget_example.png)
-
-Creating widgets that are slightly more generic and reusable are very useful and allow code and graphical modules to be shared and reused between multiple applications.
+# Estop
+`Estop` allows us to quickly and manually command physical robots to stop what they are doing. We have a couple of implementations of `Estop`, depending on which [Backend](#backend) is being used:
 
-### Miscellaneous Qt Tips
-* Define minimum and maximum sizes (if it makes sense) to help enforce the correct sizing of elements
-    * Eg. define a minimum width for a textbox based on what it's expected to contain
+* For `WifiBackend`, the `Estop` is a stateful switch connected to an Arduino. The switch state is communicated to the `WifiBackend` regularly. When `Estop` is in a `STOP` state, the `WifiBackend` overrides the [Primitives](#primitives) sent to the robot to an `Estop primitive`. when it is in a `PLAY` state, primitives are communicated normally.
```

Comment:
This isn't really true anymore. `robot_communication.py` is our AI's interface to the robots and it prevents Primives from being sent out. We support a keyboard estop as well as physical one.

Reply:
Yeah this estop section needs to be rewritten